BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/Database/EntityTypeBuilderExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Backbone.BuildingBlocks.Infrastructure.Persistence.Database;
+
+public static class EntityTypeBuilderExtensions
+{
+    public static PropertyBuilder<object> HasVersion<TEntity>(this EntityTypeBuilder<TEntity> builder, Expression<Func<TEntity, object>> propertyExpression) where TEntity : class
+    {
+        if (builder.Metadata.Model.IsNpgsql())
+        {
+            return builder.Property(propertyExpression).HasColumnType("xid").HasConversion(
+                v => (uint)v,
+                v => v
+            );
+        }
+        else
+        {
+            return builder.Property(propertyExpression).HasColumnType("rowversion").HasConversion(
+                v => (byte[])v,
+                v => v
+            );
+        }
```

Comment:
The default should be to throw an exception, in order to prevent using an unsupported column type.
```suggestion
        else if(builder.Metadata.Model.IsSqlServer())
        {
            return builder.Property(propertyExpression).HasColumnType("rowversion").HasConversion(
                v => (byte[])v,
                v => v
            );
        }
        else {
        	throw new Exception("No column type for a version has been defined for the database provider '{builder.Metadata.Model.GetDatabaseProvider}'".
        }
```